### PR TITLE
Added pinterest_test.py

### PR DESCRIPTION
**Pull request by priyankeshh**
## Description

> The changes made in this pull request should be briefly described.

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [ ] Have you adhered to the repository's defined coding convention rules?
- [ ] Have you updated the 'documentation.md' file with the method/function documentation?
- [ ] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [ ] Has the method/class been added to the documentation (md file)?

## Screenshots

> Uploading a screenshot illustrating the approach you developed for validation is required.

image
Tests Running

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

File: src/test/pinterest_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        keyword = "nature"
+        pins = self.pinterest.search_pins(keyword)
+        self.assertIsInstance(pins, list, "Expected search_pins to return a list")
+        if pins:
+            for pin in pins:
+                self.assertIn('link', pin)
+                self.assertIn('image', pin)
+
+    def test_get_pin_details(self):
+        pin_url = "https://www.pinterest.com/pin/1234567890/"  # replace with a valid Pinterest pin URL
```
Comment (nikhil25803) [inline src/test/pinterest_test.py:36]: `url = "https://www.pinterest.com/pin/1234567890/"  # replace with a valid Pinterest pin URL`

Please replace it with a valid post, then only it will be able to return desired result right? 
Make sure to test and make changes in your code accordingly instead of just pasting it from `ChatGPT`.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

**Comment by priyankeshh** [comment]
I've replaced the placeholder URLs in the test cases with valid Pinterest pin URLs and made sure the code handles potential edge cases where certain meta tags might be missing. Additionally, I tested the updated code to ensure it returns the desired results.
I apologize for the oversight in the previous commit and appreciate your guidance. Please let me know if there are any further changes needed.

image

image

image

**Comment by nikhil25803** [comment]
Great work @priyankeshh  !!
Do not forget to
+ Share your contribution on socials and tag Clueless 🫂
+ Give this repository a star ✨
+ Follow Clueless Community on GitHub ✅
